### PR TITLE
Fix compilation on macOS

### PR DESCRIPTION
**Pull request by MerryMage**
Please note: This does not enable yuzu to be playable on macOS.

### CODE REVIEW

File: src/yuzu/bootmanager.cpp [hunk 1]
```diff
@@ -571,6 +579,11 @@ bool GRenderWindow::InitializeOpenGL() {
         std::make_unique<OpenGLSharedContext>(context->GetShareContext(), child->windowHandle()));
 
     return true;
+#else
+    QMessageBox::critical(this, tr("OpenGL not available!"),
```
Comment (lioncash) [inline src/yuzu/bootmanager.cpp:583]: This should be a warning, not a critical error given this by itself isn't a stability issue

File: src/video_core/renderer_vulkan/vk_rasterizer.cpp [hunk 2]
```diff
@@ -870,7 +870,7 @@ void RasterizerVulkan::BeginTransformFeedback() {
     UNIMPLEMENTED_IF(binding.buffer_offset != 0);
 
     const GPUVAddr gpu_addr = binding.Address();
-    const std::size_t size = binding.buffer_size;
+    const VkDeviceSize size = static_cast<VkDeviceSize>(binding.buffer_size);
```
Comment (lioncash) [inline src/video_core/renderer_vulkan/vk_rasterizer.cpp:873]:
```suggestion
    const auto size = static_cast<VkDeviceSize>(binding.buffer_size);
```

File: src/video_core/renderer_vulkan/renderer_vulkan.cpp [hunk 3]
```diff
@@ -76,7 +77,8 @@ Common::DynamicLibrary OpenVulkanLibrary() {
     char* libvulkan_env = getenv("LIBVULKAN_PATH");
     if (!libvulkan_env || !library.Open(libvulkan_env)) {
         // Use the libvulkan.dylib from the application bundle.
-        std::string filename = File::GetBundleDirectory() + "/Contents/Frameworks/libvulkan.dylib";
+        std::string filename =
```
Comment (VolcaEM) [inline src/video_core/renderer_vulkan/renderer_vulkan.cpp:80]: I think this can be const

**Comment by MerryMage** [comment]
Addressed above comments.

**Pull request merged**